### PR TITLE
support aligned allocation calls with alignments < pointer size

### PR DESCRIPTION
**Pull request by christoph-cullmann**
See issue #245 

needed for e.g. code bases like LLVM

### CODE REVIEW

File: src/alloc-aligned.c [hunk 1]
```diff
@@ -14,10 +14,13 @@ terms of the MIT license. A copy of the license can be found in the file
...
 
-static void* mi_heap_malloc_zero_aligned_at(mi_heap_t* const heap, const size_t size, const size_t alignment, const size_t offset, const bool zero) mi_attr_noexcept {
+static void* mi_heap_malloc_zero_aligned_at(mi_heap_t* const heap, const size_t size, size_t alignment, const size_t offset, const bool zero) mi_attr_noexcept {
   // note: we don't require `size > offset`, we just guarantee that
   // the address at offset is aligned regardless of the allocated size.
-  mi_assert(alignment > 0 && alignment % sizeof(void*) == 0);
+
+  // if the passed alignment is too small, correct it upwards to the minimal internal required alignment
+  // fixes issues in user code calling aligned variants for types with small alignment requirements below pointer alignment
+  alignment = alignment < sizeof(void*) ? sizeof(void*) : alignment;
```
Comment (adamdmoss) [inline src/alloc-aligned.c:23]: I think this should be below the if()s below which check that we were fed with a non-0 power-of-two first, before alignment is changed.  Those requirements are sane.

**Comment by christoph-cullmann** [comment]
I updated the commit with your proposed change.
Makes sense, the consistency checks can be done first.

**Comment by daanx** [comment]
Thanks for sending a PR! We should fix this  --  however, I think just weakening the assertion is enough. With the check 
```C
 if (mi_unlikely(alignment==0 || !_mi_is_power_of_two(alignment))) return NULL;
```
we already check for powers-of-two which should satisfy the requirements. There is no need to round up the alignment itself I think since any allocation will do that automatically (as all blocks are at least `void*` aligned.

**Comment by christoph-cullmann** [comment]
The assertions got relaxed, that did solve this issue, too.

**Pull request closed without merging**